pkg/archive/jar.go

```diff
@@ -177,23 +178,40 @@ func splitPreservingQuotes(s string, sep rune) []string {
 	return result
 }
 
-var StableGitProperties = ZipEntryStabilizer{
+var StableGitProperties = ZipArchiveStabilizer{
 	Name: "jar-git-properties",
-	Func: func(zf *MutableZipFile) {
+	Func: func(zr *MutableZipReader) {
 		// These files contain git properties set by git-commit-id-maven-plugin.
 		// They contain many unreproducible attributes as documented
 		// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/825).
-		// Only JSON and properties file formats are available as documented
-		// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/95d616fc7e16018deff3f17e2d03a4b217e55294/src/main/java/pl/project13/maven/git/GitCommitIdMojo.java#L454).
-		// By default, these file are created in ${project.build.outputDirectory} and are hence at the root of the jar.
-		// We assume that the file name is 'git' as this is the default value for the plugin.
-		// We don't handle the case where the file name is changed by the user.
-		if path.Base(zf.Name) == "git.json" {
-			zf.SetContent([]byte("{}"))
+
+		// We find all the git properties files in the jar file by looking at the pom.xml files.
+		// We then remove the content of these files in order to stabilize the jar file.
+		var gitPropertiesFiles []string
+		for _, zf := range zr.File {
+			if strings.HasSuffix(zf.Name, "pom.xml") {
+				reader, err := zf.Open()
+				if err != nil {
+					continue
+				}
+				content, err := io.ReadAll(reader)
```

Comment:
hmm I guess maybe log this error? I think you'd also want to return in this case, right? we don't have a great story around bubbling up stabilizer errors. come to think of it, maybe this should just panic... we wouldn't want this to output a value we couldn't successfully stabilize.

Reply:
> maybe this should just panic... we wouldn't want this to output a value we couldn't successfully stabilize.

I agree. Having an unparseable `pom.xml` file should be notified.